features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetricDispatcher.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.rum.internal.domain.scope.RumViewManagerScope
+import com.datadog.android.rum.model.ViewEvent
+
+/**
+ * Implementation of [SessionMetricDispatcher].
+ */
+internal class SessionEndedMetricDispatcher(private val internalLogger: InternalLogger) :
+    SessionMetricDispatcher {
+
+    private val metricsMap by lazy { linkedMapOf<String, SessionEndedMetric>() }
```

Comment:
I assume that it can have less memory allocation until the metricsMap is initialised, is it justified to you?

Reply:
It will be at least the same, because `by lazy` is just a sugar around `SynchronizedLazyImpl` class which will be created to wrap a property. This class has its own fields and also involves some synchronization (for the initialization path).

In fact, in many languages containers (like map, list, etc.) are just very thin objects around the objects they store. By itself container takes just few bytes of the memory (excluding the size of the objects they keep).

So `lazy` here doesn't help. Usually it should be used for things which are heavy to initialize and they may be not needed during the class lifecycle, or for the things which cannot be initialized at the object creation time.

Reply:
Good to know, I remove all the lazy now